2018-11-07  Zhubo Deng  <classzde@lnxsrv06.seas.ucla.edu>

        * timespec.h (timespec_cmp): Fix the subtraction operator in timespec_cmp 
function with comparison operator to avoid integer overflow.

--- /u/eng/class/classzde/coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2018-11-07 00:07:49.177801000 -0800
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec)
+    return 1;
+  else if (a.tv_sec < b.tv_sec)
+    return -1;
+  else
+    return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
